qirlib/src/evaluation/runtime.rs

```diff
@@ -11,20 +11,28 @@ use log;
 use inkwell::values::FunctionValue;
 
 use super::gates::GateScope;
+use super::jit::module_functions;
 
 pub(crate) struct Simulator {
     _scope: GateScope,
 }
 
 impl<'ctx> Simulator {
-    pub fn new(module: &Module<'ctx>, ee: &ExecutionEngine<'ctx>) -> Self {
+    pub fn new(module: &Module<'ctx>, ee: &ExecutionEngine<'ctx>) -> Result<Self, String> {
         let simulator = Simulator {
             _scope: crate::evaluation::gates::GateScope::new(),
         };
 
+        if let Some(f) = Simulator::unsupported_function(module) {
```

Comment:
Can we return a list of unsupported functions as string rather than the just the first? From a user perspective it might be helpful to tell them all at once rather than making them re-run each time to find functions that need to be trimmed out.

Reply:
Done in 76361cc.